segfast/utils.py

```diff
@@ -35,6 +37,58 @@ def update(self, n=1):
 notifier = Notifier
 
 
+class TraceHeader:
+    """ Trace header class to store its name and byte position. By default, byte position is defined by name
+    accordingly to SEG-Y specification.
+
+    Parameters
+    ----------
+    name : str or int
+        If str, name of the header. If int, is interpreted as 'byte' and name will be default from spec.
+    byte : int, optional
+        Byte position of the header, by default None. If None, default byte position from the spec will be used.
+    dtype : int ot str, optional
+        dtype for header (e.g. 'i2', 'f4') or its length in bytes (then is interpreted as integer type).
+    """
+    TRACE_HEADER_SIZE = 240
+
+    def __init__(self, name, byte=None, dtype=None):
+        standard_byte_to_header = {v: k for k, v in segyio.tracefield.keys.items()}
+        start_bytes = sorted(segyio.tracefield.keys.values())
+        standard_byte_to_len = {start: end - start
+                                for start, end in zip(start_bytes, start_bytes[1:] + [self.TRACE_HEADER_SIZE + 1])}
+
+        if isinstance(name, int):
+            if byte is not None:
+                raise ValueError("'name' is int and 'byte' is defined")
+            name = standard_byte_to_header[name]
+
+        if name not in segyio.tracefield.keys:
+            warnings.warn(f'{name} is not a standard header name')
+
+        self.name = name
+        self.byte = byte or segyio.tracefield.keys[name]
+        self.standard_name = standard_byte_to_header.get(self.byte)
+        self.dtype = dtype or standard_byte_to_len[self.byte]
+        self.byte_len = self.dtype
+
+        if isinstance(self.dtype, int):
+            self.dtype = 'i' + str(self.dtype)
+
+        if isinstance(self.byte_len, str):
+            self.byte_len = int(self.byte_len[-1])
+
+        if self.byte + self.byte_len > self.TRACE_HEADER_SIZE:
+            raise ValueError(f'{self.name} header position is out of bounds')
+
+    def __eq__(self, other):
+        """ Comparison of two headers by its attributes. """
+        if isinstance(other, str):
+            return self.name == other
+
+        return self.__dict__ == other.__dict__
```

Comment:
I would rather compare them on attributes directly; also, current implementation does not show that instances with the same bytes/dtypes, but different names, are functionally same.